internal/controller/postgrescluster/postgres.go

```diff
@@ -621,6 +622,26 @@ func (r *Reconciler) reconcilePostgresDataVolume(
 
 	pvc.Spec = instanceSpec.DataVolumeClaimSpec
 
+	// If a source cluster was provided and VolumeSnapshots are turned on in the source cluster
+	// and there is a VolumeSnapshot available for the source cluster that is ReadyToUse, use it
+	// as the source for the PVC. If there is an error when retrieving VolumeSnapshots, log it,
+	// but continue creating PVC in the usual fashion.
+	if sourceCluster != nil && sourceCluster.Spec.Backups.Snapshots != nil && feature.Enabled(ctx, feature.VolumeSnapshots) {
+		snapshots, err := r.getSnapshotsForCluster(ctx, sourceCluster)
+		if err == nil {
+			snapshot := getLatestReadySnapshot(snapshots)
+			if snapshot != nil {
```

Comment:
❓ I recognize that this can happen. Is it something we should warn (event) about? 🤔 or maybe inform? (info event)

🤔 Perhaps we want the opposite: inform people when we use a snapshot? (info event)

Reply:
Maybe all of the above? If we find a good snapshot, create an Info event saying "Ready snapshot found; using snapshot to bootstrap pgdata pvc for _ cluster." If snapshots are enabled in the sourceCluster, but we don't find any snapshots, or we find one, but it isn't ReadyToUse, then we issue a Warning event.